### PR TITLE
Switch from using fractions to basisPoints

### PR DESCRIPTION
**Pull request by NickCuso**
# Description

Instead of using a numerator / denominator for values, we can simplify and define the units in basis points.  For the impacted use cases this should be plenty of precision, simplifies (for both users and the code itself) and would be a bit cheaper on gas.

Impacted transferFee and refundPenalty 

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/contracts/mixins/MixinLockCore.sol [hunk 1]
```diff
@@ -51,6 +51,9 @@ contract MixinLockCore is
   // The account which will receive funds on withdrawal
   address public beneficiary;
 
+  // The denominator component for values specified in basis points.
```
Comment (nfurfaro) [inline smart-contracts/contracts/mixins/MixinLockCore.sol]: I'm unclear why a denominator is needed if we're switching to basis points to get away from fractions?

File: smart-contracts/contracts/mixins/MixinRefunds.sol [hunk 2]
```diff
@@ -37,14 +36,13 @@ contract MixinRefunds is
...
-    uint refundPenaltyDenominator
+    uint refundPenaltyBasisPoints
   );
 
   function initialize() public
   {
-    refundPenaltyNumerator = 1;
-    refundPenaltyDenominator = 10;
+    // default to 10%
+    refundPenaltyBasisPoints = 1000;
```
Comment (nfurfaro) [inline smart-contracts/contracts/mixins/MixinRefunds.sol]: So 1000 basis points is the same as 0.1, correct?

**Comment by nfurfaro** [comment]
Why are basis points cheaper on gas?

File: smart-contracts/contracts/mixins/MixinRefunds.sol (hunk 2 above)
Comment (NickCuso) [reply smart-contracts/contracts/mixins/MixinRefunds.sol]: yup

File: smart-contracts/contracts/mixins/MixinLockCore.sol (hunk 1 above)
Comment (NickCuso) [reply smart-contracts/contracts/mixins/MixinLockCore.sol]: we want to support values < 1.  so it's still fractions under the hood, I'm just hard-coding the denominator to simplify the api

Comment (nfurfaro) [reply smart-contracts/contracts/mixins/MixinLockCore.sol]: Ahh, I think I get it. This value seems to get used as the denominator in a few calculations below, so it's just a naming thing.

**Comment by NickCuso** [comment]
Gas savings are for the our deployment and lock owner, not users.  It's less data when deploying the template and less data to update config.  But the gas savings are not significant.

I thought to do this when our team had questions about the numerator/denominator thing.  I'm hoping basis points is more natural / intuitive for people.

**Comment by NickCuso** [comment]
Oh, it is cheaper for users too!

That is because a constant is cheaper to access than a variable.

Once we get codechecks added the diff will be easier to see.

**Pull request merged**